lib/instances/LocalizationTable.lua

```diff
@@ -0,0 +1,35 @@
+local BaseInstance = import("./BaseInstance")
+local InstanceProperty = import("../InstanceProperty")
+local json = import("../json")
+
+local CONTENTS = "contents"
+
+local LocalizationTable = BaseInstance:extend("LocalizationTable")
+
+LocalizationTable.properties.SourceLocaleId = InstanceProperty.normal({
+	getDefault = function()
+		return "en-us"
+	end,
+})
+
+function LocalizationTable.prototype:SetContents(contents)
+	getmetatable(self).instance.properties[CONTENTS] = json.decode(contents)
+end
+
+function LocalizationTable.prototype:GetString(targetLocaleId, key)
+	local contents = getmetatable(self).instance.properties[CONTENTS]
+
+	if contents == nil then
+		return nil
+	end
```

Comment:
We should instead default-initialize contents to an empty table, which means we don't need to special-case this here.